test/parallel/test-stream-pipe-destory.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+require('../common');
+const { Readable, Writable } = require('stream');
+const { setImmediate } = require('node:timers/promises');
+const assert = require('assert');
+
+async function * asyncGenerator() {
+  yield '1';
+  await setImmediate();
+  yield '2';
+  await setImmediate();
+  yield '3';
+}
+
+{
+  class NullWritable extends Writable {
+    _write(c, e, cb) {
+      cb();
+    }
+  }
+
+  const src = Readable.from(asyncGenerator());
+  const dest = new NullWritable();
+  src.on('error', () => {});
+  dest.on('error', () => {});
+
+  src.pipe(dest);
+
+  process.nextTick(() => {
+    src.emit('error', new Error('Error'));
```

Comment:
```suggestion
    src.destroy(new Error('Error'));
```

This should make the test fail

Reply:
I've tested the suggested change by destroying the src.

<Details> 

```
31  - src.emit('error', new Error('Error'));
32 + src.destroy(new Error('Error'));
```

<Summary>here</Summary></Details>

But the test still passed 🤔

Reply:
I will have a look when I'm back from holiday.

Reply:
`src.destroy(new Error('Error'));` actually works, I made a mistake previously in my example and I have fixed them. I think it probably worths to add a test to document this behaviour as code?